test/common/network/connection_impl_test.cc

```diff
@@ -169,16 +172,17 @@ class ConnectionImplTest : public testing::TestWithParam<Address::IpVersion> {
     if (client_write_buffer_) {
       EXPECT_CALL(*client_write_buffer_, drain(_))
           .Times(AnyNumber())
-          .WillOnce(Invoke([&](uint64_t size) -> void { client_write_buffer_->baseDrain(size); }));
+          .WillRepeatedly(
+              Invoke([&](uint64_t size) -> void { client_write_buffer_->baseDrain(size); }));
     }
     EXPECT_CALL(client_callbacks_, onEvent(ConnectionEvent::LocalClose));
-    client_connection_->close(ConnectionCloseType::NoFlush);
     if (wait_for_remote_close) {
       EXPECT_CALL(server_callbacks_, onEvent(ConnectionEvent::RemoteClose))
           .WillOnce(Invoke([&](Network::ConnectionEvent) -> void { dispatcher_->exit(); }));
-
+      client_connection_->close(ConnectionCloseType::NoFlush);
       dispatcher_->run(Event::Dispatcher::RunType::Block);
     } else {
+      client_connection_->close(ConnectionCloseType::NoFlush);
```

Comment:
That's a byproduct of me trying to figure out why windows tests were failing. I was trying to make sure the expect is set before close is called. I suspect reverting this wouldn't be an issue.

Reply:
Finally the Windows tests are passing. Should I revert this or do we thing we might hit situation when close synchronously ends up calling onEvent?

Reply:
I can't really see how the `close()` method can call `onEvent` synchronously. It may wait for the remote close but these would require I/O loop iteration. I think you should just revert this.